lib/generator._js

```diff
@@ -264,10 +265,19 @@ ScriptGenerator.prototype.generateAspNetCoreDeploymentScript = function (_) {
     throw new Error('Only batch script files are supported for ASP.NET Core Application');
   }
 
+  if (!this.solutionPath && this.useMSBuild) {
+    throw new Error('Missing solution file path (--solutionFile). Solution file path has to be specified when used with --use-msbuild');
```

Comment:
Well, alternative design would be to not need `--use-msbuild` at all, and just use the presence of the solution as a determinant. Makes for a simpler command line, and removes some invalid combinations that we need to check for,